components/Error.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import Layout from './Layout'
+import NavLink from './NavLink'
+import { Text } from './theme/Text'
+
+type ErrorProps = {
+  type: '404' | '500'
+  message?: string
+}
+const Error: React.FC<ErrorProps> = ({ type, message }) => {
+  return (
+    <Layout>
+      <div className="container">
+        <div className="row">
+          <div className="d-flex col-sm-3 align-items-center justify-content-center">
+            <div className="text-center mt-3">
+              <Text component="div" size="xl" bold={true}>
+                {type === '404' ? 'Page not found' : 'Internal server error'}
```

Comment:
I see a high potential for someone else updating this component in the future, to add another type of error(`for example, admin pages can have an unauthorized error for non-admins`).

Therefore, I think this piece of logic can be refactored so that it is easier for others to add another error type in the future.

Right now, the next person to add onto this file may experience these difficulties:
1) Having to visually filter through `html` code to determine where to edit in the file
  * `Layout -> div -> div -> div -> div -> Text`: Have to dig all the way here to find the logic
  
2) Figuring out that `500` is the error number associated with `Internal server error`
  * With a ternary operator, it's not explicitly clear that `Internal server error` is the intended title for error `500`. A person would have to analyze the entire component as a whole, and then make the connection of `Oh, there's only one other error type, and it is error 500. So I guess Internal server error is the title for 500`.
  
3) Figuring out the best way to add a third option
  * Continuing to use a ternary operator to support a third error type makes the code unnecessarily complex and more unreadable. The next person to add another type to this file would have to go through the process of deciding how to best support a third error type.

Reply:
Your initial enum suggestion would be the best approach here in my opinion. With a enum we can directly map an error code to it's associated image, not even needing a ternary operator or switch. And it's also easy to modify in the future by adding new error status codes.

```ts
enum ErrorStatusCode {
	NOT_FOUND = '404',
	INTERNAL_SERVER_ERROR = '500'
}
```

Reply:
OK, I changed it to enum. It's easier to add new pages this way, on the other hand, using it would be easier with unions IMO since you don't need to export (and figure out extra type). But it's such a minor thing so I'm fine with either.

Reply:
Hmm I think using the `map` object makes things more complex than they have to be. `Map` is meant to be used when we have to key memory addresses such as objects, but all the status codes are just strings.

This should work:

```
const errorMessages: Readonly<{ [key in StatusCode]: string }> = {
  [StatusCode.NOT_FOUND]: 'Page not found',
  [StatusCode.INTERNAL_SERVER_ERROR]: 'Internal server error'
}
``

Reply:
Makes sense, I thought that map was faster than object for accessing values (turns out it is faster but for inserting/removing key-values pairs). It's an object now.